client/src/util/csvExport.ts

```diff
@@ -0,0 +1,192 @@
+import AnnoMatrix from "../annoMatrix/annoMatrix";
+import { Genesets } from "../reducers/genesets";
+import {
+  STANDARD_CATEGORY_NAMES,
+  EXCLUDED_CATEGORY_NAMES,
+} from "../common/types/entities";
+import { Schema, Field } from "../common/types/schema";
+import { isCategoricalAnnotation } from "./stateManager/annotationsHelpers";
+
+/**
+ * Downloads a file by creating a blob and triggering a download
+ */
+export function downloadFile(content: string, filename: string): void {
+  const blob = new Blob([content], { type: "text/csv;charset=utf-8;" });
+  const url = URL.createObjectURL(blob);
+  const a = document.createElement("a");
+  a.href = url;
+  a.download = filename;
+  a.style.display = "none";
+  document.body.append(a);
+  a.click();
+  // Revoke the blob URL and remove the element
+  setTimeout(() => {
+    URL.revokeObjectURL(url);
```

Comment:
What does revoking blob URL do? Why is there a 1 second delay for this?

What if url is revoked before download completes? Is that why there's a 1 second delay? If that's the case, this seems brittle. However, if revoking URL does not interrupt existing downloads, then there doesn't seem to be a point of the 1 second delay.